prepare_conditioning_dataset.py

```diff
@@ -3,36 +3,70 @@
 from glob import glob
 
 import torch
+from einops import rearrange
+from kornia.filters import gaussian_blur2d, laplacian
 from torch import nn
 from torch.utils.data import DataLoader, Dataset
-from torchvision.io import read_video
+from torchvision.io import read_video, write_video
 from torchvision.models.optical_flow import Raft_Large_Weights, raft_large
 from tqdm import tqdm
 
 
 class FlowPreprocessor(nn.Module):
-    out_channels: int = 2
+    out_channels: int = 3
```

Comment:
I'm not really sure how this is used, but doesn't the flow only have two channels?

Reply:
Yeah, indeed, but I spent a ton of time struggling with video pixel formats to see if there's any way to save a video with only 2 color channels. Turns out to be at least very hard if not impossible.

Instead I've just opted to add a 3rd channel which represents the flow edges (around occlusions). This allows us to save videos and maybe makes the conditioning information a bit more informative.